novelist.py

```diff
@@ -0,0 +1,187 @@
+import json
+import logging
+import requests
+import string
+import urllib
+from nose.tools import set_trace
+
+from core.coverage import CoverageProvider
+from core.metadata_layer import (
+    ContributorData,
+    IdentifierData,
+    LinkData,
+    MeasurementData,
+    Metadata,
+    SubjectData,
+)
+from core.model import (
+    DataSource,
+    Hyperlink,
+    Identifier,
+    Measurement,
+    Representation,
+    Subject,
+)
+from core.config import Configuration
+
+class NoveListAPI(object):
+
+    log = logging.getLogger("NoveList API")
+    version = "2.2"
+
+    # While the NoveList API doesn't require parameters to be passed via URL,
+    # the Representation object needs a unique URL to return the proper data
+    # from the database.
+    QUERY_ENDPOINT = "http://novselect.ebscohost.com/Data/ContentByQuery?\
+            ISBN=%(ISBN)s&ClientIdentifier=%(ClientIdentifier)s&version=%(version)s"
+    MAX_REPRESENTATION_AGE = 6*30*24*60*60      # six months
+
+    @classmethod
+    def from_config(cls, _db):
+        config = Configuration.integration(Configuration.NOVELIST_INTEGRATION)
+        profile = config.get(Configuration.NOVELIST_PROFILE)
+        password = config.get(Configuration.NOVELIST_PASSWORD)
+        if not (profile and password):
+            raise ValueError("No NoveList client configured.")
+        return cls(_db, profile, password)
+
+    def __init__(self, _db, profile, password):
+        self._db = _db
+        self.profile = profile
+        self.password = password
+        self.source = DataSource.lookup(self._db, DataSource.NOVELIST)
+
+    def lookup(self, identifier):
+        """Returns NoveList metadata for a particular identifier"""
+
+        client_identifier = identifier.urn
+        if identifier.type != Identifier.ISBN:
```

Comment:
I'd prefer to refuse to look up a non-ISBN identifier rather than try to find an equivalent. I think I know why you're doing this though. Let's talk it over.